modules/core/include/opencv2/core/mat.inl.hpp

```diff
@@ -1275,7 +1275,7 @@ Size MatSize::operator()() const
 {
     int d = dims();
     CV_DbgAssert(d <= 2);
-    return d == 2 ? Size(p[1], p[0]) : Size(p[0], 1);
+    return d == 2 ? Size(p[1], p[0]) : d == 1 ? Size(p[0], 1) : Size(0, 0);
```

Comment:
`MatSize` is only used in `cv::Mat` and `cv::UMat` right? What is the scalar you are referring to? Would a matrix of size `1x1x1x1x1x1` be a scalar ? Would size be `(1,1)` then or undefined?

An empty matrix can also be `0x1` (`dims()==2`). But when a matrix is newly created, `dims()==0`. It seems weird to also have `(0,1)`

Reply:
When empty matrix is created, dims()==0, but total() is 0 too.
But there can be scalar matrix with dims() == 0, but total() == 1:

```
Mat m, mscalar(0, nullptr, CV_32F);
mscalar.at<float>(0) = 1.f;
printf("empty m dims: %d, total number of elements: %zu, scalar m dims: %d, scalar m total: %zu\n",
       m.dims, m.total(), mscalar.dims, mscalar.total());
```

Reply:
Oh, regarding the other question. `1x1x1x1x1x1` is not a scalar, because `dims()==6`.

Reply:
Thx for the explanation. It seems the scalar case is already handled properly because `d==1` in MatSize (though `dims()==0` in cv::Mat). I added a test for that.